### PR TITLE
Update geotransolver for 2d and 3d use cases

### PR DESCRIPTION
**Pull request by coreyjadams**
# PhysicsNeMo Pull Request

This PR adds the same structured 2d and 3d convolutional steps to geotransolver that are available in transolver.

To make sure the code is not overly duplicating, it refactors a few places to ensure the implementation can reuse core components as much as possible.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: test/models/geotransolver/test_geotransolver.py [hunk 1]
```diff
@@ -556,6 +556,98 @@ def test_geotransolver_mismatched_functional_out_dims():
...
+    x3 = x4.reshape(B, H * W, 3)
+    g3 = g.reshape(B, H * W, 2)
+    y3 = model(x3, geometry=g3)
+    assert y3.shape == (B, H * W, 2)
+
+    y_none = model(x4)
+    assert y_none.shape == (B, H, W, 2)
+
+
+def test_geotransolver_structured_3d_forward(device):
```
Comment (mnabian) [inline test/models/geotransolver/test_geotransolver.py:605]: There are no tests for the `plus=True` path (not just for the new structured code, but across the entire GeoTransolver test suite.) Some of these tests could be parameterized to cover this path as well.

File: physicsnemo/experimental/models/geotransolver/context_projector.py [hunk 2]
```diff
@@ -313,6 +429,101 @@ def forward(
...
+            self.H, self.W, self.D = H, W, D_
+            self.in_project_x = nn.Conv3d(dim, inner_dim, kernel, 1, pad)
+            if not plus:
+                self.in_project_fx = nn.Conv3d(dim, inner_dim, kernel, 1, pad)
+
+        self.softmax = nn.Softmax(dim=-1)
+        self.dropout = nn.Dropout(dropout)
+        self._init_slice_components(dim_head, slice_num, heads, use_te, plus)
+
+    def _grid_project(
```
Comment (mnabian) [inline physicsnemo/experimental/models/geotransolver/context_projector.py:482]: The conv setup here mirrors PhysicsAttentionStructuredMesh2D/3D.__init__. The output layout differs (B,H,N,D vs B,N,H,D) so they can't trivially share code, but worth noting for future maintenance.

File: physicsnemo/experimental/models/geotransolver/__init__.py
Comment (mnabian) [inline physicsnemo/experimental/models/geotransolver/__init__.py:22]: Update to include the new classes

File: physicsnemo/experimental/models/geotransolver/context_projector.py
Comment (mnabian) [inline physicsnemo/experimental/models/geotransolver/context_projector.py:22]: Add `StructuredContextProjector`

**Review by mnabian — APPROVED** [review]
LGTM! Left a few minor comments.

File: physicsnemo/experimental/models/geotransolver/context_projector.py (hunk 2 above)
Comment (coreyjadams) [reply physicsnemo/experimental/models/geotransolver/context_projector.py:448]: Right - good catch.  I will update to align with PhysicsAttention's layout.

The layout in Physicsattention is designed to enable shardTensor more easily and performantly, so we're going to want this anyways long term.  Thank you.

**Review by loliverhennigh — APPROVED** [review]
I think the way nn is setup is fine

File: test/models/geotransolver/test_geotransolver.py (hunk 1 above)
Comment (coreyjadams) [reply test/models/geotransolver/test_geotransolver.py:607]: We haven't really exercised the `plus=True` path because it doesn't converge well, and the paper can't be reproduced easily.  That's why I haven't really prioritized testing it...

File: physicsnemo/experimental/models/geotransolver/context_projector.py (hunk 2 above)
Comment (coreyjadams) [reply physicsnemo/experimental/models/geotransolver/context_projector.py:448]: The current setup now shares the same conv setup via the StructureContextProjector._grid_project, so I think this has been addressed.

File: physicsnemo/experimental/models/geotransolver/context_projector.py
Comment (coreyjadams) [reply physicsnemo/experimental/models/geotransolver/context_projector.py:22]: Done.
